### ISSUE TITLE
[BUG]: Corner Case reusing type incorrectly

### ISSUE THREAD

**Issue by viceroypenguin** [thomhurst/TUnit#980]
Using the example code below, the tests will fail on successive runs of the test with different input values. The issue appears to be a combination of `[ClassDataSource<>]` and `[NotInParallel]`. 

It appears that using `[ClassDataSource<>]` does not actually use multiple instances of the class under test. When combined with `[NotInParallel]`, then the `Dispose()` becomes noticeable by later tests as the class itself has been disposed already and then reused.

Example code:
```cs
public abstract class TestBase : IDisposable
{
	private bool _disposed;

	public void CheckDisposed()
	{
		if (_disposed)
			throw new InvalidOperationException("Already disposed");
	}

	public void Dispose()
	{
		_disposed = true;
	}
}

[ClassDataSource<Dummy>]
[NotInParallel]
public sealed class Tests(Dummy dummy) : TestBase
{
	[Test]
	[MethodDataSource(nameof(GetValues))]
	public void DoTest(int value)
	{
		CheckDisposed();
	}

	public static IEnumerable<int> GetValues() => [1, 2, 3];
}

public sealed record Dummy;
```

**Comment by thomhurst** [comment]
Fix in v0.1.1099

**Comment by viceroypenguin** [comment]
Thank you!
